### ISSUE TITLE
Confusing src vs exported component names (e.g. Wrapper)

### ISSUE THREAD

**Issue by rosskevin** [dmtrKovalenko/material-ui-pickers#169]
## TL;DR
Below is an initial bug I filed, until I figured out that the exported type names don't match the source names.  More comments below this description.

-----

## Environment

| Tech                | Version |
|---------------------|---------|
| material-ui-pickers |    `master`    08891717ffa83748536a6cbd40dd2e32a5de7039 |

## Steps to reproduce
1. Look at demo
2. Look at source code propTypes

## Expected behavior
PropTypes account for anything passed in (`value`) and do not require props that are not passed in (`date`)

## Actual behavior
[Demo uses and works with `value` (example)](https://codesandbox.io/s/6wxqo5mxnz), 
```html
      <DatePicker
        value={this.state.date}
        onChange={this.handleDateChange}
      />
```
**but** propTypes require `date`, not `value` ([which does not work (example)](https://codesandbox.io/s/5w7nz7vxnl)).  Further, `value` is not even listed in `propTypes`, **and** it isn't being picked or propagated to other components - so I'm really confused as to how this works.
```js
  static propTypes = {
    date: PropTypes.object.isRequired,
    onChange: PropTypes.func.isRequired,
    autoSubmit: PropTypes.bool,
    openTo: PropTypes.oneOf(Object.keys(viewType).map(key => viewType[key])),
    disablePast: PropTypes.bool,
    disableFuture: PropTypes.bool,
    minDate: DomainPropTypes.date,
    maxDate: DomainPropTypes.date,
    showTabs: PropTypes.bool,
    leftArrowIcon: PropTypes.node,
    rightArrowIcon: PropTypes.node,
    dateRangeIcon: PropTypes.node,
    timeIcon: PropTypes.node,
    renderDay: PropTypes.func,
    utils: PropTypes.object,
    ampm: PropTypes.bool,
    shouldDisableDate: PropTypes.func,
    animateYearScrolling: PropTypes.bool,
  }
```

## Context
I'm trying to add some typescript usage tests for all the demos to start, and validate current types that exist/fix missing ones.

**Comment by rosskevin** [comment]
I spot the issue now - this is incredibly confusing since externally we are importing `DateTimePicker` yet internal source is a different class.

```
export { default as DateTimePicker } from './DateTimePicker/DateTimePickerWrapper';
```

I needed to be looking at the wrapper all along.  Perhaps some renaming might prevent confusion?

Not sure, but this stumped me as a new contributor.

**Comment by rosskevin** [comment]
Here is usage in typescript, which exemplifies the problem succinctly:

```js
import { TimePicker, DatePicker } from 'material-ui-pickers'
import { TimePickerWrapperProps } from 'material-ui-pickers/src/TimePicker/TimePickerWrapper'
import { DatePickerWrapperProps} from 'material-ui-pickers/src/DatePicker/DatePickerWrapper'
```

If an attempt to export types from the `index.d.ts` were to occur, immediately we would have a problem because `DatePickerProps !==` the props from the exported `DatePicker`.

**Comment by dmtrKovalenko** [comment]
The most cocern about that to have an ability to use not just ModalWrapper, for example PopverWrapper.

**Comment by dmtrKovalenko** [comment]
So what are you proposing - create `ModalWrapper` as a HOC and just import wrapped component in the hoc?

**Comment by rosskevin** [comment]
I would create a render callback, not a hoc, and pass the results direct into the component itself.

- Simple example of render callback (uses `children` for callback): https://react.i18next.com/components/i18n-render-prop.html#sample-usage
- react-final-form uses render callbacks extensively (uses either `render` prop or `children`): https://github.com/final-form/react-final-form
- blog post: https://cdb.reacttraining.com/use-a-render-prop-50de598f11ce

**Comment by dmtrKovalenko** [comment]
Fixed in rc-1
